### PR TITLE
Update okta.strategy.ts to work with passport-openidconnect 0.1.1

### PR DESCRIPTION
**Pull request by em-c-rod**
Closes: #5719 

To test:
1. Setup Okta development server
2. Integrated local heimdall build with Okta dev server via App integrations
3. Stored the client id, client secret, external url, and okta domain as env variables for heimdall
4. Built Heimdall using the docker build to allow for https (http does not work with the okta integration)
5. Created users in okta and assigned them to Heimdall, had to assign them a username for it to work correctly)
6. Also had to disable GP to test the integration successfully

Added documentation here for testing Okta integration - https://github.com/mitre/heimdall2/wiki/Heimdall-Authentication-Methods#testing

### CODE REVIEW

**Review by georgedias — COMMENTED** [review]
Emily, during our testing Amndeep provided instructions to update the docker builds where we used this block of code.
`RUN if curl -o install_certs.sh https://gitlab.mitre.org/mitre-scripts/mitre-pki/raw/master/os_scripts/install_certs.sh; then sh install_certs.sh centos && rm -f install_certs.sh; else echo "Not on the MII, skipping certificate install"; fi`

Do we need to add there instructions, or will the docker file be modified to contain this code?

File: apps/backend/src/authn/okta.strategy.ts [hunk 1]
```diff
@@ -35,24 +40,26 @@ export class OktaStrategy extends PassportStrategy(Strategy, 'okta') {
...
+        scope: 'openid email profile'
       },
-      (
-        _req: Request,
-        _token: string,
-        _tokenSecret: string,
-        profile: OktaProfile,
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      async function (
+        // Like in oidc.strategy.ts, we changed the arity of the function to 9 to access the data we need from 'uiProfile' due to updates in the passport-openidconnect library which otherwise caused failures in the authentication process
```
Comment (georgedias) [inline apps/backend/src/authn/okta.strategy.ts:47]: Can we please state the meaning of the underscore (_) variables, thanks.

**Review by georgedias — CHANGES_REQUESTED** [review]

**Comment by DMedina6** [comment]
> Emily, during our testing Amndeep provided instructions to update the docker builds where we used this block of code. `RUN if curl -o install_certs.sh https://gitlab.mitre.org/mitre-scripts/mitre-pki/raw/master/os_scripts/install_certs.sh; then sh install_certs.sh centos && rm -f install_certs.sh; else echo "Not on the MII, skipping certificate install"; fi`
> 
> Do we need to add there instructions, or will the docker file be modified to contain this code?

I was wondering the same thing, but I thought these changes were supposed to allow us to test behind the MITRE proxies, which isn't relevant to the customers / sponsors deploying this application.

File: apps/backend/src/authn/okta.strategy.ts (hunk 1 above)
Comment (em-c-rod) [reply apps/backend/src/authn/okta.strategy.ts:47]: Good call! Added it.

**Comment by em-c-rod** [comment]
> > Emily, during our testing Amndeep provided instructions to update the docker builds where we used this block of code. `RUN if curl -o install_certs.sh https://gitlab.mitre.org/mitre-scripts/mitre-pki/raw/master/os_scripts/install_certs.sh; then sh install_certs.sh centos && rm -f install_certs.sh; else echo "Not on the MII, skipping certificate install"; fi`
> > Do we need to add there instructions, or will the docker file be modified to contain this code?
> 
> I was wondering the same thing, but I thought these changes were supposed to allow us to test behind the MITRE proxy, which isn't relevant to the customers / sponsors deploying this application.

Correct! We don't need to add it here.

**Review by georgedias — APPROVED** [review]

**Pull request merged**